src/plugins/index_pattern_management/public/components/utils.ts

```diff
@@ -79,3 +80,31 @@ export async function getIndexPatterns(
       ) || []
   );
 }
+
+export async function getDataSources(savedObjectsClient: SavedObjectsClientContract) {
+  return (
+    savedObjectsClient
+      .find<DataSourceAttributes>({
+        type: 'data-source',
+        fields: ['title', 'type'],
+        perPage: 10000,
```

Comment:
Is there any precedence for this `10000`? Does it need to exist? Does it have to be hardcoded? Can it be configurable? Can it be read from a common config?

Reply:
+1

Reply:
it is referenced from the how other places are using this method in the repo:  https://github.com/opensearch-project/OpenSearch-Dashboards/blob/890a4b649c0a4169a8683a7ca72c0fda55e886d0/src/plugins/data_source_management/public/components/utils.ts#L14 and https://github.com/opensearch-project/OpenSearch-Dashboards/blob/890a4b649c0a4169a8683a7ca72c0fda55e886d0/src/plugins/index_pattern_management/public/components/create_index_pattern_wizard/components/step_index_pattern/step_index_pattern.tsx#L157

I am also not a fan of this approach. We can add an action item to add pagination supported client wrapper as an action item?

Reply:
`10000` is a commonly used value in the OpenSearch/osd world, it's the default max OS search hits, see reference [here](https://www.elastic.co/guide/en/elasticsearch/reference/7.10/index-modules.html#index-max-result-window). Tho it's not a good practice to use hardcode, but sadly this value is used everywhere in the codebase. My suggestion is we can add a global config such as `savedObjects.PaginationSize`, it could be tracked as a new issue

Reply:
Good to know 😊